k9mail/src/main/java/com/fsck/k9/controller/QresyncExtensionHandler.java

```diff
@@ -74,25 +82,28 @@ private void findNewRemoteMessagesToDownload(List<ImapMessage> remoteMessages, Q
 
         for (ImapMessage imapMessage : modifiedMessages) {
             long remoteMessageUid = Long.parseLong(imapMessage.getUid());
-            Message localMessage = localFolder.getMessage(imapMessage.getUid());
-
             if (remoteMessageUid < smallestLocalUid) {
                 continue;
             }
+            Message localMessage = localFolder.getMessage(imapMessage.getUid());
```

Comment:
this call loads the message including mime header parsing and stuff, when all we want is "does it not exist, or is its body not downloaded". no need to do this now, but please add a TODO comment to optimize this call